UnityProject/Assets/Scripts/Items/Weapons/Explosive.cs

```diff
@@ -85,17 +88,35 @@ private void Detonate()
 			if (isServer)
 			{
 				// Get data before despawning
-				var explosionMatrix = registerItem.Matrix;
 				var worldPos = objectBehaviour.AssumedWorldPositionServer();
 
 				// Despawn the explosive
 				_ = Despawn.ServerSingle(gameObject);
+				Explosion.StartExplosion(worldPos, explosiveStrength);
+			}
+
+		}
 
-				// Explosion here
-				var explosionGO = Instantiate(explosionPrefab, explosionMatrix.transform);
-				explosionGO.transform.position = worldPos;
-				explosionGO.Explode(explosionMatrix);
+		[Server]
+		private void AttachExplosive(GameObject target, Vector2 targetPostion)
+		{
+			if (target.TryGetComponent<PushPull>(out var handler))
+			{
+				Inventory.ServerDrop(pickupable.ItemSlot, targetPostion);
+				attachedToObject = target;
+				UpdateManager.Add(CallbackType.UPDATE,UpdateBombPosition);
```

Comment:
wut, did the tile reached not work?

Reply:
> wut, did the tile reached not work?

It didn't because it's not constantly invoked.

Reply:
why is that needed?

Reply:
Humm i guess, it is needed. Can you add a check in UpdateBombPosition for if the position of the attachedToObject is the same as the explosive to not call `SetPosition`

Reply:
Good point, will add it